.github/workflows/generative-tests.yml

```diff
@@ -46,10 +46,18 @@ jobs:
             echo
 
             # Do the actual testing
-            just test tests/generative/test_query_model.py::test_query_model
+            just test tests/generative/test_test_query_model.py
+            rc=%?
+            if [ $rc -ne 0] && [ $rc -ne 6 ]; then
+                break
+            fi
+
+            if [ $rc -eq 6 ]; then
+                echo "Dead database detected, ending test batch"
+            fi
```

Comment:
I reckon this is redundant: the test will already print the exception so it will be obvious what the failure was for someone reading the output. But maybe a comment above the `[ $rc -ne 0] && [ $rc -ne 6 ]` conditional explaining the significance of the 6 exit code would be helpful for someone reading the code.

Reply:
Righto